libpod/runtime_img.go

```diff
@@ -553,7 +553,7 @@ func getRegistriesToTry(image string, store storage.Store, defaultTransport stri
 				" the image name '%s' is incomplete.", imageName)
 		}
 		for _, searchRegistry := range searchRegistries {
-			pImage.registry = searchRegistry
+			pImage.registry = strings.TrimRight(searchRegistry, "/")
```

Comment:
Won't this remove all '/' from the end and could that cause a problem for a registry where we want at least one to remain?

Reply:
This would remove all trailing '/', yes. I am not overly familiar with the codebase yet, but I could not see any unintended side effect. In fact, now it supports configured registries that end with a slash.

A concrete example. If I set `/etc/containers/registries.conf` as follows
```toml
[registries.search]
registries = ["quay.io/foo/"]
```
and then do a `podman pull bar` I encounter the errror `error pulling image "bar": error getting default registries to try: unable to parse 'docker://quay.io/foo//bar:latest'`. That's due to the double slash between the search registry and the image. Removing the trailing slash from the search registry fixes this specific issue.

@TomSweeneyRedHat Is that fine or am I missing a side effect of the change?

Reply:
but I dont think quay.io/foo/ <-- the trailing slash is a valid registry name.

Reply:
Seems like it has the potential to be a common user error, though.
@baude What do you think about trimming trailing slashes, but printing a warning message if they exist that `example.com/foo/` is not a valid registry name and the trailing slashes should be removed?

Reply:
Well, those registries are used for more than podman ... i think it would be better to error out and say that the registry name is invalid or skip it

Reply:
@baude The semantics of the search registries are likely to lure users (at least me) to specify trailing slashes. Effectively, a search registry can be more than just the registry host, as it enables me to also add some namespace (e.g., `/foo` for `quay.io`). As they're used in a structured way, adding a slash seems intuitive. But as I've mentioned before, I am not familiar with potential side effects.

Reply:
My main concern is I wasn't sure if Podman had code down later that was expecting a slash to be at the end of the registry before it appended the image name:tag onto it.  It sounds like it doesn't so removing it here I think would be OK.  I wouldn't remove it from the file contents though in case other projects expect it to be there.

Reply:
We had a quick chat on IRC, and it seems that we need @rhatdan's eyes on the issue.

Reply:
btw, if @rhatdan says we should be handling that, it should be change in the GetRegistries code that lives in containers image

Reply:
again, if we do this, it should go here -> https://github.com/containers/image/blob/master/pkg/sysregistries/system_registries.go

Reply:
Thanks to all for looking into this :+1:  Regarding the (unintended?) search-prefix semantics: we have a concrete use-case for that with Podman (or libpod in general) and I was really happy that this works. Documentation-wise I found the comments in the `registries.conf` quite helpful, but it could be extended to mention that `registry+namespace` works as well and give a few examples.

Reply:
Open a PR, with how you would like it.  https://github.com/projectatomic/registries

Reply:
It needs to go to -> https://github.com/containers/image/blob/master/pkg/sysregistries/system_registries.go